src/tentaclio/streams/base_stream.py

```diff
@@ -98,7 +105,7 @@ class StreamerWriter(StreamBaseIO):
     connection atomic.
     """
 
-    def __init__(self, client: ContextManager[Streamer], buffer: IO):
+    def __init__(self, client: StreamerContextManager, buffer: IO):
```

Comment:
Mypy doesn't seem to recognise the `ContextManager` will enter a `Streamer`, causing non `ContextManager` methods (e.g. `put`) to through type errors. Solved by creating protocol that inherits from both `ContextManager` and `Streamer`